server/mdm/maintainedapps/ingest.go

```diff
@@ -0,0 +1,220 @@
+
+	if !strings.HasSuffix(i.baseURL, "/") {
+		i.baseURL += "/"
+	}
+
+	client := fleethttp.NewClient(fleethttp.WithTimeout(10 * time.Second))
+
+	// run at most 3 concurrent requests to avoid overwhelming the brew API
+	g.SetLimit(3)
+	for _, app := range apps {
+		app := app // capture loop variable, not required in Go 1.23+
+		g.Go(func() error {
+			return i.ingestOne(ctx, app, client)
+		})
+	}
+	return ctxerr.Wrap(ctx, g.Wait(), "ingest apps")
+}
+
+func (i ingester) ingestOne(ctx context.Context, app maintainedApp, client *http.Client) error {
+	apiURL := fmt.Sprintf("%scask/%s.json", i.baseURL, app.Identifier)
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, apiURL, nil)
+	if err != nil {
+		return ctxerr.Wrap(ctx, err, "create http request")
+	}
+
+	res, err := client.Do(req)
+	if err != nil {
+		return ctxerr.Wrap(ctx, err, "execute http request")
+	}
+	defer res.Body.Close()
+
+	body, err := io.ReadAll(res.Body)
+	if err != nil {
+		return ctxerr.Wrap(ctx, err, "read http response body")
+	}
+
+	switch res.StatusCode {
+	case http.StatusOK:
+		// success, go on
+	case http.StatusNotFound:
+		// TODO: delete the existing entry? do nothing and succeed? doing the latter for now.
+		return nil
+	default:
+		if len(body) > 512 {
+			body = body[:512]
+		}
+		return ctxerr.Errorf(ctx, "brew API returned status %d: %s", res.StatusCode, string(body))
+	}
+
+	var cask brewCask
+	if err := json.Unmarshal(body, &cask); err != nil {
+		return ctxerr.Wrapf(ctx, err, "unmarshal brew cask for %s", app.Identifier)
+	}
+
+	// validate required fields
+	if len(cask.Name) == 0 || cask.Name[0] == "" {
+		return ctxerr.Errorf(ctx, "missing name for cask %s", app.Identifier)
+	}
+	if cask.Token == "" {
+		return ctxerr.Errorf(ctx, "missing token for cask %s", app.Identifier)
+	}
+	if cask.Version == "" {
+		return ctxerr.Errorf(ctx, "missing version for cask %s", app.Identifier)
+	}
+	if cask.URL == "" {
+		return ctxerr.Errorf(ctx, "missing URL for cask %s", app.Identifier)
+	}
+	parsedURL, err := url.Parse(cask.URL)
+	if err != nil {
+		return ctxerr.Wrapf(ctx, err, "parse URL for cask %s", app.Identifier)
+	}
+	filename := path.Base(parsedURL.Path)
+
+	installScript := installScriptForApp(app, &cask)
+	uninstallScript := uninstallScriptForApp(app, &cask)
+
+	err = i.ds.UpsertMaintainedApp(ctx, &fleet.MaintainedApp{
+		Name:    cask.Name[0],
+		Token:   cask.Token,
+		Version: cask.Version,
+		// for now, maintained apps are always macOS (darwin)
+		Platform:         fleet.MacOSPlatform,
+		InstallerURL:     cask.URL,
+		Filename:         filename,
+		SHA256:           cask.SHA256,
+		BundleIdentifier: app.BundleIdentifier,
+		InstallScript:    installScript,
+		UninstallScript:  uninstallScript,
+	})
+	return ctxerr.Wrap(ctx, err, "upsert maintained app")
+}
+
+func installScriptForApp(app maintainedApp, cask *brewCask) string {
+	// TODO: implement install script based on cask and app installer format
+	return "install"
```

Comment:
Note that this will be a significant effort. See https://github.com/fleetdm/fleet/blob/7ac7ad3f9d73201789158a50f2d35b5d61caf5ef/server/mdm/maintainedapps/ingest.go#L181-L220 for a look at all the options we need to take into account (in addition to the dmg/zip extraction).